### ISSUE TITLE
Please, add a fork for wires to the game

### ISSUE THREAD

**Issue by NikitaVNsv** [transhumandesign/kag-base#1716]
Please, add a fork for wires to the game
    B
A-+-A
    B
If is A is switch on and B is swith off, A is has signal, B hasn't signal
If is A is switch off and B is swith on, A is hasn't signal, B has signal
If is A is switch on and B is swith on, A is has signal, B has signal
If is A is switch off and B is swith off, A is hasn't signal, B hasn't signal

**Comment by mugg91** [comment]
I'm rewriting MechanismsCommon.as to hopefully make it work.
Might also add a dual elbow.

**Comment by mugg91** [comment]
Sorry, I had no luck finishing it. Kept having too many errors.
When changing functions or variables inside the `MapPowerGrid` class, it said "Shared type 'MapPowerGrid' doesn't match the original declaration in other module".
If someone helps me deal with this, I can continue working on the fork tile.

**Comment by mugg91** [comment]
Basicly I would replace 
```
	array<u8> power_old;
	array<u8> power_new;
	array<u8> topo_in;      
	array<u8> topo_out;     
```
with
```
	array<array<IO_Element>> io_element_array;
```
and add
```
shared class IO_Element
{
	u8 input_topology;
	u8 output_topology;
	u8 power_new;
	u8 power_old;
	
	IO_Element(u8 input, u8 output, u8 power)
	{
		input_topology = input;
		output_topology = output;
		power_new = power;
		power_old = power;
	}
};
```

This would allow multiple input/output topologies and new/old powers in one component, since I would like to make fork work like a wire and not like a special tile.

**Comment by asumagic** [comment]
> Sorry, I had no luck finishing it. Kept having too many errors. When changing functions or variables inside the `MapPowerChunk` class, it said "Shared type 'MapPowerChunk' doesn't match the original declaration in other module". If someone helps me deal with this, I can continue working on the fork tile.

I assume you've tried it, but restarting the game may be necessary when editing shared classes, are `rebuild` is going to have a hard time with it.

**Comment by mugg91** [comment]
> I assume you've tried it, but restarting the game may be necessary when editing shared classes, are `rebuild` is going to have a hard time with it.

Nah, I didn't try. It seems that error stopped showing but now the game just doesn't load the map anymore and crashes.
I guess I will try without objects and will just use plain values to get the basics working.
